server/py/framework/db/sqldb/db.py

```diff
@@ -1958,22 +1964,26 @@ def _find_artifacts(
 
         outer_query = outer_query.join(subquery, ArtifactV2.id == subquery.c.id)
 
-        # Put "latest" tag first, then others by tag_id desc
-        latest_first_case = case(
-            (subquery.c.tag_name == "latest", 0),
-            else_=1,
-        )
-
         # join may lose order, make sure order is applied on outer as well
         # If the updated fields are the same, we need a secondary field to sort by.
-        # Third sort by tag ID to ensure consistent ordering when an artifact has multiple tags.
-        outer_query = outer_query.order_by(
-            ArtifactV2.updated.desc(),
-            ArtifactV2.id.desc(),
-            latest_first_case,
+        # If tag is not specified - sort by tag ID to ensure consistent ordering when an artifact has multiple tags.
+        if tag == "*" or not tag:
+            # Put "latest" tag first, then others by tag_id desc
+            latest_first_case = case(
+                (subquery.c.tag_name == "latest", 0),
+                else_=1,
+            )
+            # Replace the last two order criteria (which are related to tag ordering in the subquery)
+            if limit:
+                order_criteria.pop()
+                order_criteria.pop()
+            order_criteria.append(latest_first_case)
             # Safe ordering by tag_id alias
-            subquery.c[tag_id_alias].desc(),
-        )
+            order_criteria.append(subquery.c[tag_id_alias].desc())
+
+            outer_query = outer_query.order_by(*order_criteria)
+        else:
+            outer_query = outer_query.order_by(*order_criteria)
```

Comment:
It's the same in both cases
```suggestion
        outer_query = outer_query.order_by(*order_criteria)
```